tools/importer-rest-api-specs/components/dataapigenerator/helpers.go

```diff
@@ -40,20 +42,27 @@ func fileExistsAtPath(filePath string) (*bool, error) {
 	return &result, nil
 }
 
-func recreateDirectoryExcludingFiles(directory string, excludeList []string, logger hclog.Logger) error {
+func recreateDirectoryExcludingFiles(directory string, logger hclog.Logger) error {
 	logger.Trace(fmt.Sprintf("(Re)creating directory at path %q..", directory))
-	if err := os.MkdirAll(directory, os.FileMode(0755)); err != nil {
-		return fmt.Errorf("creating directory %q: %+v", directory, err)
+
+	// if the directory is an api version delete the entire directory and recreate it along with the parent service for regeneration
+	if match, err := regexp.MatchString(`/v20\d\d_\d\d_\d\d(_preview)?`, directory); match && err == nil {
```

Comment:
if we move the Service generation to before the API Version generation, we should be able to remove this and use the other `recreateDirectory` method instead?

Reply:
also worth noting, API versions unfortunately don't _have_ to be date-based (see data plane), even if this is specific to Resource Manager currently